agent/agent.go

```diff
@@ -225,6 +225,9 @@ func (a *Agent) run(ctx context.Context) {
 }
 
 func (a *Agent) handleSessionMessage(ctx context.Context, message *api.SessionMessage) error {
+	if message.Stop {
+		return a.Stop(context.Background())
```

Comment:
No, because once `<-a.stopped` blocks, we know it is open and the call to close is safe.